R/Estimation.R

```diff
@@ -43,30 +43,34 @@ getEstimation <- function(baseUrl, estimationId){
   checkmate::reportAssertions(errorMessage)
   
   url <- sprintf("%s/estimation/%d/", baseUrl, estimationId)
-
+  
   json <- httr::GET(url)
   data <- httr::content(json)
-  data$expression <- data$specification #this expression cannot be imported back - why?
   
+  if (!is.null(data$payload$message)) {
+    stop(data$payload$message)
+  }
+  
+  data$specification <- RJSONIO::fromJSON(data$specification)
   data$createdDate <- .millisecondsToDate(data$createdDate)
   data$modifiedDate <- .millisecondsToDate(data$modifiedDate)
-  data$specification <- jsonlite::fromJSON(txt = data$specification)
-  
-  data$specification$cohortDefinitions <- dplyr::as_tibble(data$specification$cohortDefinitions)
-  data$specification$conceptSets <- dplyr::as_tibble(data$specification$conceptSets)
-  data$specification$conceptSetCrossReference <- dplyr::as_tibble(data$specification$conceptSetCrossReference)
-  data$specification$negativeControls <- dplyr::as_tibble(data$specification$negativeControls)
-  
-  targetComparatorOutcomeIdDf <- list()
-  temp <- data$specification$estimationAnalysisSettings$analysisSpecification$targetComparatorOutcomes
-  
-  targetComparatorOutcomeIdDf <- data.frame(expand.grid(targetId = temp$targetId %>% unlist() %>% unique(),
-                                                        comparatorId = temp$comparatorId %>% unlist() %>% unique(),
-                                                        outcomeId = temp$outcomeIds %>% purrr::reduce(c) %>% unique())
-  )
-  data$specification$estimationAnalysisSettings$analysisSpecification$targetComparatorOutcomes <-
-    targetComparatorOutcomeIdDf
   
+  # If we don't change the R object, we can always convert it back to JSON
```

Comment:
Would the returned object be readily usable to user, or will more data wrangling needs to be done?

Reply:
We may want to help with parsing the object. e.g. many times would want to expose the parameters chosen during analysis design choices - so that it may be used in R-markdown/reports/shiny/manuscript etc.